apps/vyper/src/app/utils/remix-client.tsx

```diff
@@ -14,34 +15,70 @@ export class RemixClient extends PluginClient {
   /** Emit an event when file changed */
   async onFileChange(cb: (contract: string) => any) {
     this.client.on('fileManager', 'currentFileChanged', async (name: string) => {
-      if (!name) return
       cb(name)
     })
   }
 
+  /** Emit an event when file changed */
+  async onNoFileSelected(cb: () => any) {
+    this.client.on('fileManager', 'noFileSelected', async () => {
+      cb()
+    })
+  }
+
   /** Load Ballot contract example into the file manager */
-  async loadContract({name, content}: Contract) {
+  async loadContract({name, address}: ExampleContract) {
     try {
-      await this.client.call('fileManager', 'setFile', name, content)
-      await this.client.call('fileManager', 'switchFile', name)
+      const content = await this.client.call('contentImport', 'resolve', address)
+      await this.client.call('fileManager', 'setFile', content.cleanUrl, content.content)
+      await this.client.call('fileManager', 'switchFile', content.cleanUrl)
     } catch (err) {
       console.log(err)
     }
   }
 
+  async cloneVyperRepo() {
+    try {
+      // @ts-ignore
+      this.call('notification', 'toast', 'cloning Vyper repository...')
+      await this.call('manager', 'activatePlugin', 'dGitProvider')
+      // @ts-ignore
+      await this.call('dGitProvider', 'clone', { url: 'https://github.com/vyperlang/vyper', token: null }, 'vyper-lang')
```

Comment:
You mentioned `Vyper` at other places but creating workspace with `vyper-lang` name

Reply:
that's the issue I was discussing with with @bunsenstraat. we can't have nested folder with same name...

Reply:
ok